client/asset/eth/eth.go

```diff
@@ -372,10 +380,15 @@ func (eth *ExchangeWallet) balanceImpl() (*asset.Balance, error) {
 			fmt.Errorf("amount locked: %v > available: %v", amountLocked, gweiBal)
 	}
 
+	immature := gweiPendingBal - gweiBal
+	if gweiBal > gweiPendingBal {
+		immature = 0
```

Comment:
I'd tend to reverse the logic and use the default zero value immature, but meh, it's fine.  `var immature uint64; if gweiPendingBal < gweiBal { ...`

Reply:
I'm gathering that if you _send from_ the wallet, that debit would only be reflected in `pendingBalance` but not `balance`?

If so, that's different from dcr/btc where sending immediately reduces the spendable/available balance even if the sending tx is unconfirmed.  With dcr/btc, receiving funds at least reflects "pending" in the unconfirmed category, but if it's normal to see `gweiBal > gweiPendingBal` that must be different here.

What are your observations on `balance` vs `pendingBalance` when 1) sending from the wallet, and 2) receiving to the wallet.

I'm wondering if we need to reduce Available if immature would be negative because of a pending debit.